metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/CommonSteps.java

```diff
@@ -0,0 +1,467 @@
+    }
+
+    @Then("^the result should have (.*) as number$")
+    public void checkResponseJsonFieldValueNumber(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).isNumber());
+    }
+
+    @Then("^the result should have (.*) as null$")
+    public void checkResponseJsonFieldValueNull(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).value(nullValue()));
+    }
+
+    @Then("^the result should have (.*) as false$")
+    public void checkResponseJsonFieldValueFalse(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).value(false));
+    }
+
+    @Then("^the result should have (.*) as (.*) day$")
+    public void checkResponseJsonFieldValueDay(String field, int day) throws Exception {
+        LocalDate releaseDay;
+        if (day == 0) {
+            releaseDay = LocalDate.now();
+        } else {
+            releaseDay = LocalDate.now().plusDays(1);
+        }
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).value(releaseDay.toString()));
+    }
+
+    @Then("^the result should have (.*) existing$")
+    public void checkResponseJsonFieldValueExist(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).exists());
+    }
+
+    @Then("^the result should have (.*) non empty$")
+    public void checkResponseJsonFieldValueNotEmpty(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).isNotEmpty());
+    }
+
+    @Then("^the result should not contain (.*)$")
+    public void checkResponseJsonNoField(String field) throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$."+field).doesNotExist());
+    }
+
+    @And("^the href of the (.*) of (.*) (\\d*) should be (.*)$")
+    public void checkResponseLinkedObjectHref(String field, String className, int index, String valueKey)
+            throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$.."+className+"["+index+"].."+field+".href")
+                .value(CommonStates.getUrl(valueKey)));
+    }
+
+    @And("^the href of the class (.*) should be (.*)$")
+    public void checkResponseLinkedObjectHref(String className, String valueKey)
+            throws Exception {
+        CommonStates.getResultActions().andExpect(jsonPath("$.."+className+".href")
+                .value(CommonStates.getUrl(valueKey)));
+    }
+
+    @And("^the href list of the (.*) of (.*) (.*) contained in (.*)$")
+    public void checkResponseLinkedObjectHrefListLoop(String field, String className, int lenght, String valueKey)
+            throws Exception {
+        String[] urlList = valueKey.split(",");
+        for (int index = 0; index < lenght; index++) {
+            CommonStates.getResultActions().andExpect(jsonPath("$.."+className+"["+index+"].."+field+".href")
```

Comment:
This is not safe. Please remove the possibilities of out of bound access occuring.

Reply:
This is no longer an issue as it has been removed in 5aaf388.